haskell-best-practices.md

```diff
@@ -13,6 +13,67 @@
      x `divEx` d = x / d
  ```
 
+### Avoid `MonadFail`
+
+Don't write our own functions using `MonadFail`/`fail`.
```

Comment:
Should `fail` be considered a restricted keyword in our hlint configuration?

Reply:
Sadly there are some libraries which utilize `MonadFail` as their only point of failure. Ex: `cassava`.

Reply:
Can we change this to:
> Don't write functions constrained by `MonadFail`.